server/auth/gatekeeper.go

```diff
@@ -337,11 +337,13 @@ func addClaimsLogFields(claims *types.Claims, fields log.Fields) log.Fields {
 	return fields
 }
 
-func DefaultClientForAuthorization(authorization string) (*rest.Config, *servertypes.Clients, error) {
+func DefaultClientForAuthorization(authorization string, config *rest.Config) (*rest.Config, *servertypes.Clients, error) {
 	restConfig, err := kubeconfig.GetRestConfig(authorization)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to create REST config: %w", err)
 	}
+	restConfig.QPS = config.QPS
+	restConfig.Burst = config.Burst
```

Comment:
I mentioned above, but I think it might be better to merge the configs together. For instance, the Server's `rest.Config` also has [a user agent](https://github.com/argoproj/argo-workflows/blame/42262690fcbab0ed6106d3f9069e128655b3339f/cmd/argo/commands/server.go#L81).

Those are the only 3 configurations right now, so maybe 1 line for each is fine, but a merge could be more resilient to change if other configurations are added to the Server